Backend/api/user/controller.js

```diff
@@ -115,10 +118,82 @@ const deleteUser = async (req, res, next) => {
   }
 };
 
+// Method used to get all workouts
+const getWorkouts = async (req, res, next) => {
+  try {
+    let user = await User.findById(req.user.payload._id).populate("workouts");
+
+    if (!user) {
+      return next({ message: "The user was not found." });
+    }
+
+    workouts = user.workouts;
+    res.status(200).json({ workouts });
+  } catch (error) {
+    return next(error);
+  }
+};
+
+// Method used to create a workout
+const createWorkout = async (req, res, next) => {
+  try {
+    let user = await User.findById(req.user.payload._id);
+    let workout = new Workout(req.body);
+
+    if (!user) {
+      return next({ message: "The user was not found." });
+    }
+
+    const sess = await mongoose.startSession();
+    sess.startTransaction();
+    await workout.save({ session: sess });
+    user.workouts.push(workout);
+    await user.save({ session: sess });
+    await sess.commitTransaction();
+
+    res.status(200).json({ status: "The workout was created." });
+  } catch (error) {
+    return next(error);
+  }
+};
+
+// Method used to delete a workout
+const deleteWorkout = async (req, res, next) => {
+  try {
+    let user = await User.findById(req.user.payload._id);
+
+    if (!user) {
+      return next({ message: "The user was not found." });
+    }
+
+    let workout = user.workouts.find(
+      (workout) => workout._id == req.params._workoutId
+    );
+
+    if (!workout) {
+      return next({ message: "The workout was not found." });
+    }
+
+    const sess = await mongoose.startSession();
+    sess.startTransaction();
+    await book.findByIdAndDelete({ session: sess });
```

Comment:
If I'm not mistaken, `workout` should be used instead of `book`.
Also, not sure if `findByIdAndDelete` works on a document instance, probably `remove` is the method you want.